src/transfers-history/services/SpokePoolEventsQueryService.ts

```diff
@@ -0,0 +1,98 @@
+import { TypedEvent } from "@across-protocol/contracts-v2/dist/typechain/common";
+import { FundsDepositedEvent, FilledRelayEvent } from "@across-protocol/contracts-v2/dist/typechain/SpokePool";
+import { BigNumber, providers } from "ethers";
+
+import { TransfersRepository } from "../adapters/db/transfers-repository";
+import { Logger } from "../adapters/logger";
+import { ISpokePoolContractEventsQuerier } from "../adapters/web3";
+import { ChainId } from "../adapters/web3/model";
+import { clientConfig } from "../config";
+import { Transfer } from "../model";
+
+export class SpokePoolEventsQueryService {
+  private latestBlockNumber: number | undefined;
+
+  constructor(
+    private chainId: ChainId,
+    private provider: providers.Provider,
+    private eventsQuerier: ISpokePoolContractEventsQuerier,
+    private logger: Logger,
+    private transfersRepository: TransfersRepository,
+    private depositorAddr?: string
+  ) {}
+
+  public async getEvents() {
+    const from = this.latestBlockNumber
+      ? this.latestBlockNumber + 1
+      : clientConfig.spokePools[this.chainId].lowerBoundBlockNumber;
+    const to = (await this.provider.getBlock("latest")).number;
+
+    if (from > to) {
+      this.logger.debug(
+        "[SpokePoolEventsQueryService::getEvents]",
+        `🔴 chain ${this.chainId}: from ${from} > to ${to}`
+      );
+      return;
+    };
+
+    this.logger.debug(
+      "[SpokePoolEventsQueryService::getEvents]",
+      `🟢 chain ${this.chainId}: fetched events from ${from} to ${to}`
+    );
+    const depositEvents = await this.eventsQuerier.getFundsDepositEvents(from, to, this.depositorAddr);
+    const filledRelayEvents = await this.eventsQuerier.getFilledRelayEvents(from, to, this.depositorAddr);
+    this.logger.debug(
+      "[SpokePoolEventsQueryService::getEvents]",
+      `🟢 chain ${this.chainId}: fetched ${depositEvents.length} FundsDeposited events and ${filledRelayEvents.length} FilledRelayEvents`
+    );
+    const blockTimestampMap = await this.getBlocksTimestamp([...depositEvents, ...filledRelayEvents]);
```

Comment:
im really surprised no timestamp data is available on the events, also why is timestamp so important?

Reply:
we already discussed this during the meeting. It was the only criteria I found for sorting events coming from multiple chains